py/vttest/local_database.py

```diff
@@ -94,20 +95,25 @@ def load_schema(self):
 
     if not self.schema_dir:
       return
-    for keyspace in os.listdir(self.schema_dir):
+    
+    if not os.path.isdir(self.schema_dir):
+      raise Exception('schema_dir "%s" is not a directory.' % self.schema_dir)
+    
+    for keyspace in set([shard.keyspace for shard in self.shards]):
       keyspace_dir = os.path.join(self.schema_dir, keyspace)
-      if os.path.isdir(keyspace_dir):
-        for filename in os.listdir(keyspace_dir):
-          filepath = os.path.join(keyspace_dir, filename)
-          logging.info(
-              'Loading schema for keyspace %s from file %s', keyspace,
-              filepath)
-          cmds = self.get_sql_commands_from_file(filepath, keyspace_dir)
-
-          # Run the cmds on each shard in the keyspace.
-          for shard in self.shards:
-            if shard.keyspace == keyspace:
-              self.mysql_execute(cmds, db_name=shard.db_name)
+      if not os.path.isdir(keyspace_dir):
+        raise Exception('No keyspace subdirectory found in the schema dir. '
```

Comment:
There might be a test case where you'd create a keyspace without any initial schema. Can we make this log a warning and continue to the next keyspace?